radarr/movieeditor.go

```diff
@@ -0,0 +1,69 @@
+package radarr
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"golift.io/starr"
+)
+
+const bpMovieEditor = bpMovie + "/editor"
+
+// BulkEdit is the input for the bulk movie editor endpoint.
+// You may use starr.True(), starr.False(), starr.Int64(), and starr.String() to add data to the struct members.
+type BulkEdit struct {
+	MovieIDs            []int64 `json:"movieIds"`
+	Monitored           *bool   `json:"monitored,omitempty"`
+	QualityProfileID    *int64  `json:"qualityProfileId,omitempty"`
+	MinimumAvailability *string `json:"minimumAvailability,omitempty"` // tba
+	RootFolderPath      *string `json:"rootFolderPath,omitempty"`      // path
+	Tags                []int   `json:"tags,omitempty"`                // [0]
+	ApplyTags           *string `json:"applyTags,omitempty"`           // add
+	MoveFiles           *bool   `json:"moveFiles,omitempty"`
+	DeleteFiles         *bool   `json:"deleteFiles,omitempty"`        // delete only
+	AddImportExclusion  *bool   `json:"addImportExclusion,omitempty"` // delete only
```

Comment:
are they pointer to let them leverage the `omitempty` tag and use the default 0 value as well?
when they are not present which is the default behaviour? if it is the same as the 0 value you can use values instead of pointers and remove the `omitempty` tag.
I haven't tried this, so they're just some thoughts

Reply:
These members are purposely pointers, because the default value is different than the "missing" value.

This is a bulk-change endpoint. When you pick `monitored`, for instance, there are three choices. `monitored`, `not monitored` and `no change`. We achieve "no change" by omitting the key entirely. If we set it to `true`, then all movie IDs provided are monitored. If we set it to `false` then they are all unmonitored.  If we do not want to change the monitored status, but rather, change something else, then we have to omit `monitored`.

Reply:
got it, thanks for the example